app/helpers/photo_helper.rb

```diff
@@ -0,0 +1,108 @@
+module PhotoHelper
+
+  def search_photos rover, params
+    photos = rover.photos.order(:camera_id, :id).search params, rover
+
+    if params[:page]
+      photos = photos.page(params[:page]).per params[:per_page]
+    end
+
+    photos
+  end
+
+  class InvalidSizeParameter < StandardError
+    def initialize(
+        size,
+        rover_name,
+        exception_type="custom"
+      )
+      super("Invalid size parameter '#{size}' for #{rover_name.titleize}")
+    end
+  end
+
+  def resize_photo photo, params
+    rover_name = photo.rover.name.downcase
+    suffix_data = lookup_suffix rover_name, params[:size]
+
+    if !suffix_data.nil?
+      replace_photo_suffix photo, suffix_data[:old_length], suffix_data[:new]
+    else
+      raise InvalidSizeParameter.new params[:size], rover_name
+    end
+  end
+
+  def resize_photos photos, params
+    rover_name = params[:rover_id].downcase
+    suffix_data = lookup_suffix rover_name, params[:size]
+
+    if suffix_data.present?
+      replace_each_photo_suffix photos, suffix_data[:old_length], suffix_data[:new]
+    else
+      raise InvalidSizeParameter.new params[:size], rover_name
+    end
+  end
+
+  private
+
+  def replace_each_photo_suffix photos, old_suffix_length, new_suffix
+    photos.map do |photo|
+      replace_photo_suffix photo, old_suffix_length, new_suffix
+    end
+  end
+
+  def replace_photo_suffix photo, old_suffix_length, new_suffix
+    new_photo = photo.clone;
+    new_photo.img_src = photo.img_src[0, photo.img_src.length - old_suffix_length] + new_suffix
```

Comment:
This certainly works, but an alternative would be to use `photo.image_src.gsub(old_suffix, new_suffix)`.